client/auth/login.jsx

```diff
@@ -1,80 +1,90 @@
 /** @format */
+
 /**
  * External dependencies
  */
 import React from 'react';
-import createReactClass from 'create-react-class';
-import { localize } from 'i18n-calypso';
 import Gridicon from 'gridicons';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
+import AuthCodeButton from './auth-code-button';
+import AuthStore from 'lib/oauth-store';
 import config from 'config';
-import Main from 'components/main';
-import FormTextInput from 'components/forms/form-text-input';
-import FormPasswordInput from 'components/forms/form-password-input';
-import FormFieldset from 'components/forms/form-fieldset';
 import FormButton from 'components/forms/form-button';
 import FormButtonsBar from 'components/forms/form-buttons-bar';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormPasswordInput from 'components/forms/form-password-input';
+import FormTextInput from 'components/forms/form-text-input';
+import LostPassword from './lost-password';
+import Main from 'components/main';
 import Notice from 'components/notice';
-import AuthStore from 'lib/oauth-store';
-import * as AuthActions from 'lib/oauth-store/actions';
-import eventRecorder from 'me/event-recorder';
-import WordPressLogo from 'components/wordpress-logo';
-import AuthCodeButton from './auth-code-button';
 import SelfHostedInstructions from './self-hosted-instructions';
-import LostPassword from './lost-password';
-
-export const Login = createReactClass( {
-	displayName: 'Auth',
-	mixins: [ eventRecorder ],
-
-	componentDidMount: function() {
+import WordPressLogo from 'components/wordpress-logo';
+import { login } from 'lib/oauth-store/actions';
+import { recordGoogleEvent } from 'state/analytics/actions';
+
+export class Auth extends React.Component {
+	state = Object.assign(
```

Comment:
I tend to avoid `Object.assign` in favor of using the spread operator for cases like this these days:

```js
state = {
	login: '',
	password: '',
	auth_code: '',
	...AuthStore.get()
};
```

Reply:
Good call, fixed in 6bb40f06a505c9c0be2dc49abb6c493dbd33b35b.